metrics/write.go

```diff
@@ -225,13 +267,25 @@ func (c *Client) write(ctx context.Context) error {
 				if end > len(tss) {
 					end = len(tss)
 				}
-				req := &prompb.WriteRequest{
-					Timeseries: tss[i:end],
+				var req R
+				switch ts := any(tss).(type) {
+				case []*writev2.TimeSeries:
+					req = any(&writev2.Request{
+						Timeseries: ts[i:end],
+						Symbols:    st.Symbols(),
+					}).(R)
```

Comment:
this looks scary xD

Reply:
Yeah was experimenting. Generics work, but result in super tricky code. I'll revert back to what I had initially